packages/viewer/src/lib/rendering/canvas.ts

```diff
@@ -118,17 +119,23 @@ export function createCanvasRenderer(): CanvasRenderer {
     const frameNumber = data.frame.sequenceNumber;
     const frameIsNewer =
       lastFrameNumber == null || frameNumber > lastFrameNumber;
-    const image = await loadImageBytes(data.frame.image.imageBytes);
-    const predicatePassing = data.predicate?.() ?? true;
+    const loadPredicatePassing = data.loadPredicate?.() ?? true;
+
+    if (frameIsNewer && loadPredicatePassing) {
+      const image = await loadImageBytes(data.frame.image.imageBytes);
+      const drawPredicatePassing = data.drawPredicate?.() ?? true;
```

Comment:
Drawing an older frame is a bug with the renderer IMO. The default implementation shouldn't allow this, so I don't see a need of a predicate here. If we did introduce a predicate later, I'd suggest the predicate is not even called for a frame that is older.

I would suggest that we:
* Break up the decoding of an image from the drawing. Basically treat them as two different processes within the renderer.
* Don't block on image decoding, e.g. no `await`. It'll try to decode as fast as it can, and forward the decoded image to the drawing process. The decoding process will still not attempt to decode old frames. Because decoding is async, it's possible that images are forwarded to the drawing process out of order.
* The drawing process takes the decoded images and is aware of the last drawn frame. It will not attempt to draw a decoded image that is older.